node/node.go

```diff
@@ -292,6 +295,29 @@ func (n *Node) run(ctx context.Context) (err error) {
 				}
 				n.Unlock()
 				renewCert()
+			case newRoot := <-n.notifyRootCAChange:
+				n.Lock()
+				role := n.role
+				n.Unlock()
+				if role == ca.ManagerRole || bytes.Equal(newRoot, securityConfig.RootCA().Certs) {
+					continue
+				}
+				newRootCA, err := ca.NewRootCA(newRoot, nil, nil, ca.DefaultNodeCertExpiration, nil)
+				if err != nil {
+					log.G(ctx).WithError(err).Error("invalid new root certificate from the dispatcher")
+					continue
+				}
+				// TODO(cyli): make sure it validates against our current TLS certificate first
+				if err := securityConfig.UpdateRootCA(&newRootCA, newRootCA.Pool); err != nil {
+					log.G(ctx).WithError(err).Error("could not use new root CA from dispatcher")
+					continue
+				}
+				newIssuer := securityConfig.IssuerInfo()
+				n.notifyTLSChange <- &api.NodeTLSInfo{
+					CertIssuerPublicKey: newIssuer.PublicKey,
+					CertIssuerSubject:   newIssuer.Subject,
+					TrustRootHash:       newRootCA.Digest.String(),
+				}
```

Comment:
I get worried about blocking writes like this that aren't wrapped in a `select`. We've had a few bugs on shutdown where writes to channels can hang. In this case, I think there will be problems if the agent shuts down at the moment we try to write on this channel. Let's make sure there's a strategy to ensure that either a channel write is guaranteed to have a goroutine receiving it, or there's another case in a `select` that handles the receiver going away.